src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.cs

```diff
@@ -210,6 +217,11 @@ private DeclarationModifiers CreateMethodModifiers()
                 var isStatic = !this.AnalyzerResult.UseInstanceMember;
                 var isReadOnly = this.AnalyzerResult.ShouldBeReadOnly;
 
+                if (LocalFunction && !this.Options.GetOption(CSharpCodeStyleOptions.PreferStaticLocalFunction).Value)
+                {
+                    isStatic = false;
```

Comment:
💡 I wonder if PreferStaticLocalFunction should also mean that if instance members or locals are used, we generate a static local but with them passed as a parameter.

Reply:
Great idea. 👍  I believe locals are already passed into the method 100% of the time by default, but I think having instance members passed as well would be a good addition. Would this have to go through design review?

Reply:
Not a bad idea to make sure that the behavior of "PreferStaticLocalFunction" makes sense still with that change.